src/std/fs.ab

```diff
@@ -108,3 +108,32 @@ pub fun glob_multiple(paths: [Text]): [Text]? {
 pub fun glob(path: Text): [Text]? {
     return glob_multiple([path])?
 }
+
+/// Extract the file detecting from the filename the extension
+/// Supports: bz2, gz, xz, bz2, deb, rar, rpm, tar(gz/xz/bz), zip(war/jar), 7z
+/// Note: Not all the commands supports the output folder path
+pub fun extract(path: Text, target: Text): Null? {
+    if file_exist(path) {
+        if {
+            match_regex(path, "\.\(tar\.bz2\|tbz\|tbz2\)$"): $ tar xvjf "{path}" -C "{target}" $?
+            match_any_regex(path, ["tar.gz$", "tgz$"]): $ tar xzf "{path}" -C "{target}" $?
```

Comment:
You need to apply the same change to each of these function calls, including the ones with only a single regex. There's no need to call `match_any_regex` at all here.